### PR TITLE
make corehq/apps/userreports compatible with python 3

### PR DESCRIPTION
**Pull request by nickpell**
@emord @millerdev

### CODE REVIEW

**Comment by nickpell** [comment]
✅

File: corehq/apps/userreports/models.py [hunk 1]
```diff
@@ -395,7 +397,7 @@ def map_col(column):
                     'XFormInstance': _('Forms'),
                     'CommCareCase': _('Cases')
                 }.get(self.config.referenced_doc_type, "Layer"),
-                'columns': filter(None, [map_col(col) for col in self.columns])
+                'columns': ifilter(None, [map_col(col) for col in self.columns])
```
Comment (emord) [inline corehq/apps/userreports/models.py]: I think you want `list(filter(None, ...))` here. I'm not if we guarantee that this could handle an iterator

Comment (millerdev) [inline corehq/apps/userreports/models.py]: `futurize` would convert this to a list comprehension. I think that should be our standard approach.

Comment (emord) [inline corehq/apps/userreports/models.py]: May wan to keep the filter if `map_col` is expensive though

Comment (millerdev) [inline corehq/apps/userreports/models.py]: or (preferably) do this
```py
[x for x in (map_col(col) for col in self.columns) if x]
```

Comment (nickpell) [inline corehq/apps/userreports/models.py]: > ```[x for x in (map_col(col) for col in self.columns) if x]```

nice, I like this!

**Comment by nickpell** [comment]
✅

**Pull request merged**